### PR TITLE
Adding simple cloudbuild file that builds, tags, and pushes the docker image

### PR DESCRIPTION
**Pull request by kfswain**

### CODE REVIEW

**Comment by kfswain** [comment]
Pairs with: https://github.com/kubernetes/test-infra/pull/33930

**Comment by kfswain** [comment]
Fixes #55

File: cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+steps:
```
Comment (ahg-g) [inline cloudbuild.yaml:1]: fyi, lws and jobset are slightly different: https://github.com/kubernetes-sigs/lws/blob/main/cloudbuild.yaml

not an expert on this file, so I am not sure if we need to do the same thing here.

Comment (kfswain) [reply cloudbuild.yaml:1]: Yeah, they are using `make` as their build command. We are just using a simple `docker build` for now, so we are using a base image. I expect we will increase in sophistication as time goes on but this works for now I think. I've tested this on my personal GCP project's cloud build (I had to change the registry, but thats just permissioning issues)

**Pull request merged**